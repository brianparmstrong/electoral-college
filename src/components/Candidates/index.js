import React from 'react';
import { arrayOf, number, shape, string } from 'prop-types';
import Candidate from '../Candidate';

const Candidates = ({
  candidatesData,
  popularVoteTotals,
  winnerTakeAllTotals
}) => {
  const candidates = candidatesData.map((candidate, i) => {
    return (
      <Candidate
        name={candidate.name}
        key={`candidate-${i}`}
        party={candidate.party}
        popularVoteTotals={popularVoteTotals[i]}
        winner={candidate.winner}
        winnerTakeAllTotal={winnerTakeAllTotals[i]} />
    );
  });
  return (
    <div className='candidatesWrapper'>{candidates}</div>
  );
};

Candidates.propTypes = {
  candidatesData: arrayOf(shape({
    name: string.isRequired,
    party: string.isRequired
  })).isRequired,
  popularVoteTotals: arrayOf(number).isRequired,
  winnerTakeAllTotals: arrayOf(number).isRequired
};

export default Candidates;
