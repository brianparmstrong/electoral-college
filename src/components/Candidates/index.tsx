import Candidate from '../Candidate';
import { IfcCandidates } from '../../types';

const Candidates = (candidates: IfcCandidates) => {
  const { candidatesData, popularVoteTotals, winnerTakeAllTotals } = candidates;
  const candidatesMap = candidatesData.map((candidate, i) => (
    <Candidate
      name={candidate.name}
      key={`candidate-${i}`}
      party={candidate.party}
      popularVoteTotals={popularVoteTotals[i]}
      winnerTakeAllTotal={winnerTakeAllTotals[i]}
    />
  ));
  return <div className="candidatesWrapper">{candidatesMap}</div>;
};

export default Candidates;
