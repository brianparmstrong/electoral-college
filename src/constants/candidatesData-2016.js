export const CANDIDATES_DATA = [
  {
    "name": "Trump",
    "party": "gop"
  },
  {
    "name": "Clinton",
    "party": "dem" 
  },
  {
    "name": "Johnson",
    "party": "lib"
  },
  {
    "name": "Stein",
    "party": "grn"
  },
  {
    "name": "McMullin",
    "party": "ind"
  }
];
