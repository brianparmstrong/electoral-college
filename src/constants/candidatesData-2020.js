export const CANDIDATES_DATA = [
  {
    "name": "Trump",
    "party": "gop"
  },
  {
    "name": "Biden",
    "party": "dem" 
  },
  {
    "name": "Jorgensen",
    "party": "lib"
  },
  {
    "name": "Hawkins",
    "party": "grn"
  },
  {
    "name": "West",
    "party": "ind"
  }
];
