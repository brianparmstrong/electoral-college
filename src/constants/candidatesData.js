export const CANDIDATES_DATA = [
  {
    "name": "Rep",
    "party": "gop"
  },
  {
    "name": "Dem",
    "party": "dem" 
  },
  {
    "name": "Lib",
    "party": "lib"
  },
  {
    "name": "Grn",
    "party": "grn"
  },
  {
    "name": "Ind",
    "party": "ind"
  }
];
