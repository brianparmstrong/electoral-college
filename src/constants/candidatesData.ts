import { CandidatesData } from '../types';

export const CANDIDATES_DATA: CandidatesData = [
  {
    name: 'Trump',
    party: 'gop',
  },
  {
    name: 'Biden',
    party: 'dem',
  },
  {
    name: 'Oliver',
    party: 'lib',
  },
  {
    name: 'Stein',
    party: 'grn',
  },
  {
    name: 'Kennedy',
    party: 'ind',
  },
];
