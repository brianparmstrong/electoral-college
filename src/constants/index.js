import { CANDIDATES_DATA } from './candidatesData-2024';
import { PARTY_MAP } from './partyData';
import { STATES_DATA } from './statesData-2020';

export {
  CANDIDATES_DATA,
  PARTY_MAP,
  STATES_DATA
};
