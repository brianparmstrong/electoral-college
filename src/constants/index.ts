import { PARTY_MAP } from './partyData';
import { STATES_DATA } from './statesData';

export { PARTY_MAP, STATES_DATA };
