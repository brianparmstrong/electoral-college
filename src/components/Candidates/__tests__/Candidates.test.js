import React from 'react';
import renderer from 'react-test-renderer';
import { CANDIDATES_DATA } from '../../../constants';
import Candidates from '../';

const MOCK_POP_VOTE_TOTALS = [ 40, 40, 2, 1, 1 ];
const MOCK_WINNER_TAKE_ALL_TOTALS = [ 43, 42, 0, 0, 0 ];

it('renders correctly', () => {
  const candidates = renderer
    .create(
      <Candidates
        candidatesData={CANDIDATES_DATA}
        popularVoteTotals={MOCK_POP_VOTE_TOTALS}
        winnerTakeAllTotals={MOCK_WINNER_TAKE_ALL_TOTALS} />
    )
    .toJSON();
  expect(candidates).toMatchSnapshot();
});
