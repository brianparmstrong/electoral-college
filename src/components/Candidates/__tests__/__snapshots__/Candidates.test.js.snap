// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="candidatesWrapper"
>
  <div
    className="candidate"
    id="Trump"
  >
    <h2>
      Trump
    </h2>
    <img
      alt="Trump"
      src="trump.jpeg"
    />
    <div
      className="electoralVotesWrapper"
    >
      <p
        className="electoralVotesLabel"
      >
        Winner-Take-All:
      </p>
      <span
        className="electoralVotes"
        id="gop-wta"
      >
        43
      </span>
      <p
        className="electoralVotesLabel"
      >
        Proportional:
      </p>
      <span
        className="electoralVotes"
        id="gop-prop"
      >
        40
      </span>
    </div>
  </div>
  <div
    className="candidate"
    id="Biden"
  >
    <h2>
      Biden
    </h2>
    <img
      alt="Biden"
      src="biden.jpeg"
    />
    <div
      className="electoralVotesWrapper"
    >
      <p
        className="electoralVotesLabel"
      >
        Winner-Take-All:
      </p>
      <span
        className="electoralVotes"
        id="dem-wta"
      >
        42
      </span>
      <p
        className="electoralVotesLabel"
      >
        Proportional:
      </p>
      <span
        className="electoralVotes"
        id="dem-prop"
      >
        40
      </span>
    </div>
  </div>
  <div
    className="candidate"
    id="Jorgensen"
  >
    <h2>
      Jorgensen
    </h2>
    <img
      alt="Jorgensen"
      src="jorgensen.png"
    />
    <div
      className="electoralVotesWrapper"
    >
      <p
        className="electoralVotesLabel"
      >
        Winner-Take-All:
      </p>
      <span
        className="electoralVotes"
        id="lib-wta"
      >
        0
      </span>
      <p
        className="electoralVotesLabel"
      >
        Proportional:
      </p>
      <span
        className="electoralVotes"
        id="lib-prop"
      >
        2
      </span>
    </div>
  </div>
  <div
    className="candidate"
    id="Hawkins"
  >
    <h2>
      Hawkins
    </h2>
    <img
      alt="Hawkins"
      src="hawkins.jpeg"
    />
    <div
      className="electoralVotesWrapper"
    >
      <p
        className="electoralVotesLabel"
      >
        Winner-Take-All:
      </p>
      <span
        className="electoralVotes"
        id="grn-wta"
      >
        0
      </span>
      <p
        className="electoralVotesLabel"
      >
        Proportional:
      </p>
      <span
        className="electoralVotes"
        id="grn-prop"
      >
        1
      </span>
    </div>
  </div>
  <div
    className="candidate"
    id="West"
  >
    <h2>
      West
    </h2>
    <img
      alt="West"
      src="west.jpeg"
    />
    <div
      className="electoralVotesWrapper"
    >
      <p
        className="electoralVotesLabel"
      >
        Winner-Take-All:
      </p>
      <span
        className="electoralVotes"
        id="ind-wta"
      >
        0
      </span>
      <p
        className="electoralVotesLabel"
      >
        Proportional:
      </p>
      <span
        className="electoralVotes"
        id="ind-prop"
      >
        1
      </span>
    </div>
  </div>
</div>
`;
